src/encord_active/lib/project/project_file_structure.py

```diff
@@ -31,116 +35,229 @@
-                            location=label_row.label_row_file.as_posix(),
-                        )
-                    )
-
-                if fill_data_units:
-                    data_units = label_row_dict["data_units"]
-                    for data_unit in label_row.iter_data_unit():
-                        du = data_units[data_unit.hash]
-                        if data_type == "video":
-                            if "_" not in data_unit.path.stem:
-                                frame_str = "-1"  # To include a reference to the video location in the DataUnit table
-                            else:
-                                _, frame_str = data_unit.path.stem.rsplit("_", 1)
-                        else:
-                            frame_str = du.get("data_sequence", 0)
-                        frame = int(frame_str)
-
-                        batcher.dataunit.create(
-                            DataUnitCreateInput(
-                                data_hash=data_unit.hash,
-                                data_title=du["data_title"],
-                                frame=frame,
-                                location=data_unit.path.as_posix(),
-                                lr_data_hash=lr_data_hash,
-                            )
-                        )
-            batcher.commit()
-
-
 class DataUnitStructure(NamedTuple):
-    hash: str
-    path: Path
+    label_hash: str
+    du_hash: str
+    data_type: str
+    # If type is 'video' this is the video link not the image link
+    signed_url: str
+    # Video frame or sequence index
+    frame: Optional[int]
+    # Raw data_hash
+    data_hash_raw: str
+    # Data unit metadata
+    width: int
+    height: int
+    frames_per_second: float
 
 
 class LabelRowStructure:
-    def __init__(self, path: Path, mappings: dict[str, str]):
-        self.path: Path = path
+    def __init__(self, mappings: dict[str, str], label_hash: str, project: "ProjectFileStructure"):
         self._mappings: dict[str, str] = mappings
         self._rev_mappings: dict[str, str] = {v: k for k, v in mappings.items()}
+        self._label_hash = label_hash
+        self._project = project
 
     def __hash__(self) -> int:
-        return hash(self.path.as_posix())
+        return hash(self._label_hash)
+
+    def label_row_file_deprecated_for_migration(self) -> Path:
+        return self._project.project_dir / "data" / self._label_hash / "label_row.json"
 
     @property
-    def label_row_file(self) -> Path:
-        return self.path / "label_row.json"
+    def label_row_json(self) -> Dict[str, Any]:
+        with PrismaConnection(self._project) as conn:
+            entry = conn.labelrow.find_unique(where={"label_hash": self._label_hash})
+            label_row_json = "missing"
+            if entry is not None:
+                label_row_json = entry.label_row_json
+            return json.loads(label_row_json)
```

Comment:
[nitpick] 💅

This block will always throw an exception when `entry is not None` since `"missing"` isn't a valid json 🙈